### ISSUE TITLE
I have YoloV5 working Python script. How can I upgrade it to YoloV7 or YoloV8 for better accuracy?

### ISSUE THREAD

**Issue by FurkanGozukara** [ultralytics/yolov5#11993]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

```
import cv2
import torch
import os
from PIL import Image

# Load the pre-trained YOLOv5 model
model = torch.hub.load('ultralytics/yolov5', 'yolov5x')

# Aspect ratios to consider
aspect_ratios = [(1024, 1280)]

def auto_zoom(input_dir, output_base_dir):
    # Loop over all files in the input directory
    for filename in os.listdir(input_dir):
        # Create the full input path and read the file
        input_path = os.path.join(input_dir, filename)
        img = cv2.imread(input_path)

        if img is None:
            continue

        # Run the image through the model
        results = model(img)

        # Find the first human detected in the image
        human = next((x for x in results.xyxy[0] if int(x[5]) == 0), None)

        if human is None:
            print(f"No human detected in the image {input_path}.")
            os.remove(input_path)
            continue

        # Crop the image to the bounding box of the human
        x1, y1, x2, y2 = map(int, human[:4])
```

**Comment by glenn-jocher** [comment]
@FurkanGozukara hi there!

It's great to hear that you have YOLOv5 up and running in your Python script. YOLOv5 is currently the latest version available. Our team is consistently working to enhance the model and its accuracy, so I recommend keeping an eye on our [documentation](https://docs.ultralytics.com/yolov5) for any updates or new releases.

Keep up the good work, and let us know if you have any other questions or need further assistance!

**Comment by FurkanGozukara** [comment]
> @FurkanGozukara hi there!
> 
> It's great to hear that you have YOLOv5 up and running in your Python script. YOLOv5 is currently the latest version available. Our team is consistently working to enhance the model and its accuracy, so I recommend keeping an eye on our [documentation](https://docs.ultralytics.com/yolov5) for any updates or new releases.
> 
> Keep up the good work, and let us know if you have any other questions or need further assistance!

interesting

there is also yolo v7 and yolo v8?

actually i managed to upgrade yolo v7 working great : https://youtu.be/Fbuyu35TkE4

**Comment by glenn-jocher** [comment]
@FurkanGozukara that's really interesting! While YOLOv7 and YOLOv8 are not official versions maintained by the Ultralytics team, it's great to hear that you've been able to experiment and upgrade the model. Your video showcasing YOLOv7 in action is impressive!

If you have any specific questions or need support with your YOLOv7 implementation, please feel free to ask. 🚀

**Comment by FurkanGozukara** [comment]
> @FurkanGozukara that's really interesting! While YOLOv7 and YOLOv8 are not official versions maintained by the Ultralytics team, it's great to hear that you've been able to experiment and upgrade the model. Your video showcasing YOLOv7 in action is impressive!
> 
> If you have any specific questions or need support with your YOLOv7 implementation, please feel free to ask. 🚀

thank you so much. the thing is i couldn't manage to make it work with yolo v8

it is not same way supported i think. do you know that model?

by the way yolo v7 made huge upgrade to my process

**Comment by glenn-jocher** [comment]
@FurkanGozukara You're welcome! I'm glad to hear that YOLOv7 has had a positive impact on your workflow.

As of now, YOLOv8 is not an official version maintained by our team, so I don't have specific information regarding its implementation. However, it's great to see your enthusiasm for exploring and experimenting with different versions. If there are any specific questions or issues you encounter with YOLOv8, feel free to reach out and I'll do my best to help. Keep up the great work!

**Comment by FurkanGozukara** [comment]
> @FurkanGozukara You're welcome! I'm glad to hear that YOLOv7 has had a positive impact on your workflow.
> 
> As of now, YOLOv8 is not an official version maintained by our team, so I don't have specific information regarding its implementation. However, it's great to see your enthusiasm for exploring and experimenting with different versions. If there are any specific questions or issues you encounter with YOLOv8, feel free to reach out and I'll do my best to help. Keep up the great work!

ty so much

here my code can you help me to make it work with yolo v8?

```
    with counter_lock:
        model = torch.hub.load('WongKinYiu/yolov7', 'custom', 'yolov7-e6e.pt', force_reload=False, trust_repo=True)
```

```
            with model_lock:
                results = model(img_path)
            detections = results.pandas().xyxy[0]

            person_detected = detections[detections['name'] == 'person']
            if not person_detected.empty:
                x1, y1, x2, y2 = person_detected.iloc[0][['xmin', 'ymin', 'xmax', 'ymax']].astype(int)
```


so I downloaded yolov7-e6e.pt file, loading it, making detections  and filtering with detections[detections['name'] == 'person']

how to do same with yolo v8?

**Comment by glenn-jocher** [comment]
@FurkanGozukara Absolutely, I can guide you through the update. However, it's important to note that "yolov7-e6e.pt" is specific to YOLOv7 and may not be directly compatible with YOLOv8. Also, since YOLOv8 is not an official version maintained by our team, I recommend checking the official repository or documentation for YOLOv8 for the appropriate model files and usage.

For YOLOv8, you would need to follow the specific loading and detection procedures outlined in the official documentation or repository. It's also advised to refer to the YOLOv8 documentation and examples for implementing the similar detection process you have with YOLOv7.

If you encounter any specific issues or have further questions during the transition to YOLOv8, feel free to reach out. Good luck with your implementation, and keep up the great work!
